vispy/visuals/filters/mesh.py

```diff
@@ -177,6 +177,7 @@ def _attach(self, visual):
         vec3 u = dFdx(v_pos_scene.xyz);
         vec3 v = dFdy(v_pos_scene.xyz);
         normal = cross(u, v);
+    } else {
         // Note(asnt): The normal calculated above always points in the
         // direction of the camera. Reintroduce the original orientation of the
         // face.
```

Comment:
I guess this note is referring to the normal calculation in the vertex shader, not the cross product done in the `if` block right above it? So that makes me (someone who doesn't understand this much at all), should this frontfacing check be dedented and be applied to flat shading and smooth shading? Obviously your video looks pretty good, but I don't see a difference between flat and smooth.

Reply:
>  should this frontfacing check be de-dented and be applied to flat shading and smooth shading

That's what I thought at first, but if you do that then the `flat` shading gets flipped. Apparently, what we have in main is the worst combination of all, because `smooth` needs to be flipped when inside-out, but not `flat`.

I won't pretend I understand exactly why this is happening, so I'm basing my code on the visual result :P

Reply:
Ok. I did a quick test with the `mesh_shading.py` example and this branch and it seems to look the same so at least no obvious issues with any other cases. I guess we'll just go with :ok_hand: and see if anyone complains. @asnt if you're around and have any memory on your note and this code and if this fix seems correct it'd be great to get your insight.

@brisvag feel free to merge if you're OK with this.